hoot-services/src/main/java/hoot/services/controllers/osm/map/FolderResource.java

```diff
@@ -0,0 +1,369 @@
+import static hoot.services.models.db.QMaps.maps;
+import static hoot.services.utils.DbUtils.createQuery;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.querydsl.core.types.dsl.Expressions;
+import com.querydsl.sql.SQLQuery;
+
+import hoot.services.models.db.FolderMapMappings;
+import hoot.services.models.db.Folders;
+import hoot.services.models.db.Users;
+
+
+@Controller
+@Path("/api/0.6/map/folders")
+@Transactional
+public class FolderResource {
+    private static final Logger logger = LoggerFactory.getLogger(FolderResource.class);
+
+    /**
+     * Returns a list of all folders in the services database.
+     *
+     * GET hoot-services/osm/api/0.6/map/folders
+     *
+     * @return a JSON object containing a list of folders
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public FolderRecords getFolders(@Context HttpServletRequest request) {
+        Users user = Users.fromRequest(request);
+        FolderRecords folderRecords = null;
+
+        SQLQuery<Folders> sql = createQuery()
+                .select(folders)
+                .from(folders);
+        if(user != null) {
+            sql.where(
+                    folders.userId.eq(user.getId()).or(folders.publicCol.isTrue())
+            );
+        }
+        List<Folders> folderRecordSet = sql.orderBy(folders.displayName.asc()).fetch();
+        folderRecords = mapFolderRecordsToFolders(folderRecordSet);
+        return folderRecords;
+    }
+
+    /**
+     * Returns a list of all folders in the services database
+     *
+     * GET hoot-services/osm/api/0.6/map/folders/linked
+     *
+     * @return a JSON object containing a list of folders
+     */
+    @GET
+    @Path("/linked")
+    @Produces(MediaType.APPLICATION_JSON)
+    public LinkRecords getLinks(@Context HttpServletRequest request) {
+        Users user = Users.fromRequest(request);
+        LinkRecords linkRecords = null;
+
+        createQuery().delete(folderMapMappings)
+            .where(new SQLQuery<>().from(maps).where(folderMapMappings.mapId.eq(maps.id)).notExists())
+            .execute();
+
+        try {
+            createQuery().insert(folderMapMappings).columns(folderMapMappings.mapId, folderMapMappings.folderId)
```

Comment:
This function is a bit weird, it contains a **maintenance operation** -- which should probably be converted to foreign key table constraints w/ `ON DELETE CASCADE`.

+ The delete statement removed orphaned folder mapping resources in cases where maps were deleted, but their link was not.
+ The insert statement does the opposite, adding folder links to new maps that haven't been linked to the root folder, `0`. 

From the user's prospective the function is a GET because no changes should be occurring from their request.

Scope Creep Enhancement:
+ Database migration
    + foreign key, ensure integrity of the three way relationship between `maps`, `folders`, and `folder_map_mappings`
+ Refactoring
    + Anywhere new `maps` are created, ensure creation of `folder_map_mappings` with default `folder_id` of `0` to comply with table constraints
    + verify `folder` / `map` removal logic